filechooser-portal/Main.vala

```diff
@@ -209,32 +203,34 @@ public class Files.FileChooserPortal : Object {
             }
         }
 
-        try {
-            dialog.register_id = connection.register_object<Xdp.Request> (handle, dialog);
-        } catch (Error e) {
-            critical (e.message);
-        }
+        dialog.register_object (connection, handle); // Dialog will unregister itself when disposed
 
         var _results = new HashTable<string, Variant> (str_hash, str_equal);
         uint _response = 2;
 
-        dialog.destroy.connect (() => {
-            if (dialog.register_id != 0) {
-                connection.unregister_object (dialog.register_id);
-            }
-        });
-
         dialog.response.connect ((id) => {
             switch (id) {
                 case Gtk.ResponseType.OK:
-                    _results["uris"] = dialog.get_uris ();
-                    _results["choices"] = dialog.get_choices ();
-                    if (dialog.filter != null) {
-                        _results["current_filter"] = dialog.filter.to_gvariant ();
+                    var chosen_uri = dialog.get_uri ();
+                    if (chosen_uri != supplied_uri) { // No need to check full uri supplied by calling app
+                        check_overwrite_uri.begin (chosen_uri, (obj, res) => {
+                            if (check_overwrite_uri.end (res)) {
+                                _results["uris"] = dialog.get_uris ();
+                                _results["choices"] = dialog.get_choices ();
+                                if (dialog.filter != null) {
+                                    _results["current_filter"] = dialog.filter.to_gvariant ();
+                                }
+
+                                _response = 0;
+                                warning ("Save File callback");
```

Comment:
```suggestion
```
no need for this here, or should be downgraded to debug.

Reply:
Yes - that was just for testing.